subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c

```diff
@@ -43,25 +44,83 @@ static int tp_udp_init(struct mqtt_sn_transport *transport)
 {
 	struct mqtt_sn_transport_udp *udp = UDP_TRANSPORT(transport);
 	int err;
+	struct sockaddr addrm;
+	struct ip_mreqn mreqn;
+	int optval;
+	struct net_if *iface;
 
-	udp->sock = zsock_socket(udp->gwaddr.sa_family, SOCK_DGRAM, 0);
+	udp->sock = zsock_socket(udp->bcaddr.sa_family, SOCK_DGRAM, 0);
 	if (udp->sock < 0) {
 		return errno;
 	}
 
 	LOG_DBG("Socket %d", udp->sock);
 
-#ifdef LOG_DBG
-	char ip[30], *out;
+	optval = 1;
+	err = zsock_setsockopt(udp->sock, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
+	if (err < 0) {
+		return errno;
+	}
+
+	if (IS_ENABLED(CONFIG_MQTT_SN_LOG_LEVEL_DBG)) {
+		char ip[30], *out;
+		uint16_t port;
+
+		out = get_ip_str((struct sockaddr *)&udp->bcaddr, ip, sizeof(ip));
+		switch (udp->bcaddr.sa_family) {
+		case AF_INET:
+			port = ntohs(((struct sockaddr_in *)&udp->bcaddr)->sin_port);
+			break;
+		case AF_INET6:
+			port = ntohs(((struct sockaddr_in6 *)&udp->bcaddr)->sin6_port);
+			break;
+		default:
+			break;
+		}
+
+		if (out != NULL) {
+			LOG_DBG("Binding to Brodcast IP %s:%u", out, port);
+		}
+	}
+
+	switch (udp->bcaddr.sa_family) {
+	case AF_INET:
+		addrm.sa_family = AF_INET;
+		((struct sockaddr_in *)&addrm)->sin_port =
+			((struct sockaddr_in *)&udp->bcaddr)->sin_port;
+		((struct sockaddr_in *)&addrm)->sin_addr.s_addr = INADDR_ANY;
```

Comment:
Is IPv4 always enabled, if it is, then we should have `depends on NET_IPV4` in Kconfig file.
Otherwise we should place `IS_ENABLED(CONFIG_NET_IPV4)` here like in IPv6.